File: week1_C/pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long n=0,tmp=0;
    int cont=0,sum1=0,sum2=0;

    do
    {
        n = get_long("Number: ");
    } while (n < 1);
    
    tmp = n;

    while(tmp>0)
    {
        sum1 = sum1 + tmp%10;
        tmp = tmp/10;
        cont++;

        if(tmp>0)
        {
            if(tmp%10 < 5)
            {
                sum2 = sum2 + tmp%10*2;
            }
            else{
                sum2 = sum2 + tmp%10*2%10 + tmp%10*2/10;
            }
            tmp = tmp/10;
            cont++;
        }
        
    }

    if((sum1+sum2)%10 == 0)
    {
        if((cont==13 && n/(long)pow(10,12)==4) || (cont==16 && n/(long)pow(10,15)==4))
        {
            printf("VISA\n");
        }
        else if(cont==15 && (n/(long)pow(10,13)==34 || n/(long)pow(10,13)==37))
        {
            printf("AMEX\n");
        }
        else if(cont==16 && (n/(long)pow(10,14)>=51 || n/(long)pow(10,14)<=55))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
        
    }
    else
    {
        printf("INVALID\n");
    }
    

}

File: week6_Python/pset6/credit.py
from cs50 import get_int,get_string

def main():
    n = get_string("Number: ")

    # a = check(n)
    # print(a)
    # print(n[0:2])

    if check(n) and (len(n)==13 or len(n)==16) and int(n[0])==4:
        print("VISA")
    elif check(n) and len(n)==15 and (int(n[0:2])==34 or int(n[0:2])==37):
        print("AMEX")
    elif check(n) and len(n)==16 and (int(n[0:2])==51 or int(n[0:2])==55):
        print("MASTERCARD")
    else:
        print("INVALID")


def check(n):
    alter = n[::-1]
    sum1 = 0
    sum2 = 0
    cont = 0

    for i in alter:
        if cont%2 == 1:
            if int(i)>=5:
                sum1 = sum1 + int(str(int(i)*2)[0]) + int(str(int(i)*2)[1])
            else:
                sum1 = sum1 + int(i)*2
        else:
            sum2 = sum2 + int(i)
        cont = cont+1

    if (sum1+sum2)%10 ==0:
        return True
    else:
        return False

main()